File: solutions/2000/main.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        word_list = list(word)
        stack = []
        for w in word_list: 
            stack.append(w)
            if w == ch: 
                n = len(stack)
                for i in range(n): 
                    word_list[i] = stack.pop()
                return "".join(word_list) 
        return word 

if __name__ == "__main__": 
    word = "abcdefd"
    ch = "d"
    s = Solution()
    assert s.reversePrefix(word, ch) == "dcbaefd"

File: solutions/2000/main.cpp
#include <string> 
#include <stack> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    string reversePrefix(string word, char ch) {
        stack<char> st; 
        for (const auto& w : word) {
            st.push(w); 
            if (w == ch) {
                const int n = st.size(); 
                for (int i = 0; i < n; ++i) {
                    word[i] = st.top(); 
                    st.pop(); 
                }
                return word; 
            }
        }
        return word; 
    }
};

TEST(Test2000, SimpleTest) {
    string word = "abcdefd"; 
    char ch = 'd'; 
    Solution s; 
    EXPECT_EQ(s.reversePrefix(word, ch), "dcbaefd"); 
}

int main() {
    testing::InitGoogleTest(); 
    return RUN_ALL_TESTS(); 
}
